lib/ably/models/channel_details.rb

```diff
@@ -1,59 +1,65 @@
-module Ably::Models
-  # Convert token details argument to a {ChannelDetails} object
-  #
-  # @param attributes (see #initialize)
-  #
-  # @return [ChannelDetails]
-  def self.ChannelDetails(attributes)
-    case attributes
-    when ChannelDetails
-      return attributes
-    else
-      ChannelDetails.new(attributes)
-    end
-  end
+# frozen_string_literal: true
 
-  # ChannelDetails is a type that represents information for a channel including channelId, name, status and occupancy (CHD1)
+module Ably
+  # Models module provides the methods and classes for the Ably library
   #
-  class ChannelDetails
-    extend Ably::Modules::Enum
-    extend Forwardable
-    include Ably::Modules::ModelCommon
-
-    # The attributes of ChannelDetails (CHD2)
+  module Models
+    # Convert token details argument to a {ChannelDetails} object
     #
-    attr_reader :attributes
-
-    alias_method :to_h, :attributes
-
-    # Initialize a new ChannelDetails
+    # @param attributes (see #initialize)
     #
-    def initialize(attrs)
-      @attributes = IdiomaticRubyWrapper(attrs.clone)
+    # @return [ChannelDetails]
+    def self.ChannelDetails(attributes)
```

Comment:
I haven't seen this `self.<module name>` syntax before – would you mind explaining what it does, and what is the violation that it’s replacing?

Reply:
@lawrence-forooghian 

The yardoc comment was added by me. The definition was not changed. I don't use this syntax but there are some more similar definitions in the gem.

Reply:
Automatic fix from `rubocop`, generated invalid code so most changes were made manually.
`standardrb --fix` were made in last commit https://github.com/ably/ably-ruby/pull/373/commits/a7971921cd1ddb2ab41894c29c1ec0bd04a0123a

Reply:
@lawrence-forooghian it's a classic approach that is used when you want to convert to object to object of desired class unless it's already of the desired class. ruby uses it `Kernel::Float` `Kernel::Array`